File: Angular/src/app/Services/receptionist.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Patient } from '../Components/Patient';
import {Appointments} from '../Components/Appointments';

@Injectable({
  providedIn: 'root'
})
export class ReceptionistService {
  
  private baseUrl ="https://localhost:7199/api/Receptionist"

  constructor(private http:HttpClient){

  }

  getPatients(): Observable<any> {
    return this.http.get(`${this.baseUrl}/GetPatients`);
  }

  getSpecilization(): Observable<any> {
    return this.http.get(`${this.baseUrl}/GetSpecilizations`);
  }

  GetDoctorsBySpecialization(Specl: string): Observable<any> {
    return this.http.get(`${this.baseUrl}/GetDoctorsBySpecilization`, {
      params: { Specialization: Specl }
    });
  }

  GetAvailDatesByDoctor(DocId : number) :  Observable<any> {
    return this.http.get(`${this.baseUrl}/GetAvailDatesByDoctor`, {
      params: { doc: DocId }
    });
  }

  AddPatient(Pat : Patient) : Observable<any>{
    return this.http.post(`${this.baseUrl}/AddPatient`,Pat);
  }

  AddAppointment(App : Appointments) : Observable<any>{
    return this.http.post(`${this.baseUrl}/ScheduleAppointment`,App);
  }

  GetAllDoctorAppointments() : Observable<any>{
    return this.http.get(`${this.baseUrl}/GetAllDoctorAppointments`);
  }

  CancelAppointment(id : number) :  Observable<any> {
    return this.http.put(`${this.baseUrl}/CancelAppointment/${id}`, null);
  }

  getPatientById(id : number): Observable<any> {
    return this.http.get(`${this.baseUrl}/GetPatientById/${id}`);
  }

  editPatient(id:number,Pat : Patient):Observable<any>{
    return this.http.put(`${this.baseUrl}/UpdatePatient/${id}`,Pat);
  }

  DeletePatient(id:number):Observable<any>{
    return this.http.delete(`${this.baseUrl}/DeletePatient/${id}`);
  }

  GetPatientRecordsAfterReview() : Observable<any> {
    return this.http.get(`${this.baseUrl}/GetPatientRecordsAfterReview`);
  }

  genarateInvoice(Pid : number,Aid : number): Observable<any> {
    return this.http.post(`${this.baseUrl}/GenerateInvoice/${Pid}/${Aid}`,'');
  }

}

File: .net core/ClinicManagement/Service/ReceptionistService.cs
using ClinicManagement.Interface;
using ClinicManagement.Model;

namespace ClinicManagement.Service
{
    public class ReceptionistService
    {
        private readonly IReceptionist rec;

        public ReceptionistService(IReceptionist rec)
        {
            this.rec = rec;
        }

        public async Task<IEnumerable<Patient>> GetPatients()
        {
            return await rec.GetPatients();
        }

        public async Task AddPatient(Patient patient)
        {
            await rec.AddPatient(patient);
        }

        public async Task UpdatePatient(int id, Patient patient)
        {
            await rec.UpdatePatient(id, patient);
        }

        public async Task<Patient> GetPatientById(int id)
        {
            return await rec.GetPatientById(id);
        }

        public async Task<IEnumerable<string>> GetSpecilizations()
        {
            return await rec.GetSpecilizations();
        }

        public async Task<IEnumerable<DoctorDto>> GetDoctorsBySpecilization(string Specialization)
        {
            return await rec.GetDoctorsBySpecialization(Specialization);
        }

        public async Task DeletePatient(int id)
        {
            await rec.DeletePatient(id);
        }

        public async Task<IEnumerable<DateTime>> GetAvailDatesByDoctor(int doc)
        {
            return await rec.GetAvailDatesByDoctor(doc);
        }

        public async Task ScheduleAppointment(Appointment appointment)
        {
            await rec.ScheduleAppointment(appointment);
        }

        public async Task CancelAppointment(int appointmentId)
        {
            await rec.CancelAppointment(appointmentId);
        }

        public async Task<IEnumerable<AllAppointmentDto>> GetAllDoctorAppointments()
        {
            return await rec.GetAllDoctorAppointments();
        }

        public async Task<IEnumerable<object>> GetPatientRecordsAfterReview()
        {
            return await rec.GetPatientRecordsAfterReview();
        }

        public async Task generateInvoice(int pid, int aid)
        {
            await rec.generateInvoice(pid, aid);
        }

    }
}
